enso/experiment/rationalized.py

```diff
@@ -211,14 +214,10 @@ def fit(self, X, Y):
                 rationales.append([{**label, "label": l[1]} for label in l[0]])
             else:
                 rationales.append([])
-        rationale_texts = [
-            rationale['text'] 
-            for doc in rationales 
-            for rationale in doc
-        ]
-        docs = np.asarray([self.NLP(str(x), disable=['ner', 'tagger', 'textcat']) for x in X])
-        rationale_docs = np.asarray([self.NLP(rationale) for rationale in rationale_texts if len(rationale)])
-        self._train_rationale_model(docs, rationale_docs)
+        rationale_texts = [rationale["text"] for doc in rationales for rationale in doc]
+        docs = np.asarray([self.NLP(str(x), disable=["ner", "tagger", "textcat"]) for x in X])
```

Comment:
can we do this with NLP.pipe?

Reply:
ooh good call! just tested and it does give a speedup in both training and predict. Will change in enso now, not sure about how that translates into production when we get to it.